miniapp/io.cpp

```diff
@@ -389,6 +409,64 @@ std::ostream& operator<<(std::ostream& o, const cl_options& options) {
     return o;
 }
 
+
+/// Parse spike times from a stream
+/// A single spike per line, white space is ignored
+/// All characters after a parsed floating point nr on a line are ignored
+/// Throws a usage error when parsing fails
+///
+/// Returns a unique_ptr to a vector of time_type
+
+std::unique_ptr<std::vector<time_type> > parse_spike_times_from_stream(
+    std::ifstream & fid) {
```

Comment:
This function is overly complicated, and will actually fail on some cases, e.g. in the following file:
```
1 2.3
3.2 2.3
```
The output will be `[12.3, error]`, when you would expect `[1, 3.2]`.

Here is a simpler and safer approach:
```
std::vector<time_type> times;
std::string line;                                                                                        
while (std::getline(fid, line)) {
    std::stringstream s(line);

    time_type t;
    s >> t >> std::ws;

    if (!s || s.peek()!=EOF) {
        throw std::runtime_error(                                                                        
            util::sprintf(
                "Unable to parse spike file on line %d: \"%s\"\n",
                times.size(), line));
        }
    }

    times.push_back(t);
}
```